blockly-core/blocks/functional_procedures.js

```diff
@@ -224,6 +224,19 @@ Blockly.Blocks.functional_definition = {
       this.updateParams_();
     }
   },
+  changeParamType: function(name, newType) {
+    var change = false;
+    for (var x = 0; x < this.parameterNames_.length; x++) {
+      if (Blockly.Names.equals(name, this.parameterNames_[x])) {
+        this.parameterTypes_[x] = newType;
+        change = true;
+      }
+    }
```

Comment:
Do we expect multiple entries in `parameterNames_` to match `name`?  If not, can we break out of this loop after we find one match?